subsys/lorawan/lorawan.c

```diff
@@ -599,6 +610,10 @@ static int lorawan_init(const struct device *dev)
 		return -EINVAL;
 	}
 
+#if !defined(CONFIG_LORAWAN_NVM_NONE)
```

Comment:
I  believe the preferred pattern (when possible) is:
```
if (!IS_ENABLED(CONFIG_LORAWAN_NVM_NONE)) {
    lorawan_nvm_data_restore();
}
```

Reply:
done